ios/Classifier/VCPTaxonomy.m

```diff
@@ -174,14 +175,26 @@ - (NSDictionary *)aggregateScores:(MLMultiArray *)classification currentNode:(VC
 
         if ([leafScore floatValue] >= self.taxonomyRollupCutoff) {
             allScores[node.taxonId] = leafScore;
+        } else {
+            self.excludedLeafScoreSum += leafScore.floatValue;
         }
     }
 
     return [allScores copy];
 }
 
-- (NSDictionary *)aggregateScores:(MLMultiArray *)classification {
-    return [self aggregateScores:classification currentNode:self.life];
+- (NSDictionary *)aggregateAndNormalizeScores:(MLMultiArray *)classification {
+    // Reset the sum of removed leaf scores
+    self.excludedLeafScoreSum = 0.0;
+    NSDictionary *scores = [self aggregateScores:classification currentNode:self.life];
```

Comment:
Do I have to remove this dict after I am finished with re-normalization.

Reply:
nope, it'll get deallocated after you leave this scope as long as there is no other active reference to it. the objc runtime doesn't have a garbage collector, but if you squint hard enough, ARC kinda looks like a gc.